### ISSUE TITLE
Ports: `libmpg123` won't compile

### ISSUE THREAD

**Issue by iProgramMC** [SerenityOS/serenity#14060]
So I was trying to compile freedink for use within Serenity and it depends on `SDL2_mixer`, which itself depends on `libmpg123`. All goes good until it finds an error in the ./configure (??) file. [Here's a gist link to the output](https://gist.github.com/iProgramMC/29ed0d2747b9566c4273da9466b5d753)

**Comment by iProgramMC** [comment]
![image](https://user-images.githubusercontent.com/17717494/169023074-83c832e8-6ef5-4742-a920-378f77f5f5c3.png)
I have not been able to find a file with the error at that line, though I do suspect it's actually an issue with the configure script. 
![image](https://user-images.githubusercontent.com/17717494/169023254-dee938ff-684d-482f-a2b5-aa082d9aafc1.png)
![image](https://user-images.githubusercontent.com/17717494/169023360-bdac949a-ea42-42db-bf68-d1e62e8f6ba9.png)

**Comment by timschumi** [comment]
It seems like you have a broken version of `autoconf`, the part about `PKG_CHECK_MODULES` is supposed to be replaced, it shouldn't end up in the configure script verbatim.

**Comment by iProgramMC** [comment]
Ok, so I did `sudo apt update && sudo apt upgrade`, got autoconf V2.69. Tried rebuilding libmpg123 and got this error:
```
//heck of a lot more above
       (derived from LIBS, only used for end-user binaries and modules)
  LIBDL='-ldl'
  LIBMPG123_LIBS='-lm '
  LIBSYN123_LIBS='-lm'
  LIBOUT123_LIBS='-lm '

Next type 'make' and then 'make install'.
Building libmpg123...
+ make -j8
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /home/iprogramincpp-ub2/serenity/Ports/libmpg123/mpg123-1.29.3/build/missing aclocal-1.16 -I m4
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /home/iprogramincpp-ub2/serenity/Ports/libmpg123/mpg123-1.29.3/build/missing autoconf
 cd . && /bin/bash /home/iprogramincpp-ub2/serenity/Ports/libmpg123/mpg123-1.29.3/build/missing automake-1.16 --gnu
configure.ac:276: error: possibly undefined macro: LT_SYS_MODULE_EXT
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
make: *** [Makefile:2939: configure] Error 1
make: *** Waiting for unfinished jobs....
iprogramincpp-ub2@IPROGRAMINCPP:~/serenity/Ports/libmpg123$
```

**Comment by BertalanD** [comment]
This looks interesting, and gives a couple clues about the problem. So `make` (!) is calling `automake` to regenerate `configure`. Mind you, this is after `configure` has already been executed once, since that's how the Makefile is created in the first place.

The reason `LT_SYS_MODULE_EXT` is missing is likely because you have a different version of Libtool than what libmpg123 expects, and the two versions are not compatible. But that should not be a problem, because there is no real need to regenerate `configure`. We need to figure out why `make` wants to regenerate it.

**Comment by iProgramMC** [comment]
@BertalanD I have the latest version that it'll let me get:
```
iprogramincpp-ub2@IPROGRAMINCPP:~$ sudo apt install libtool
Reading package lists... Done
Building dependency tree
Reading state information... Done
libtool is already the newest version (2.4.6-14).
The following package was automatically installed and is no longer required:
  libfwupdplugin1
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
iprogramincpp-ub2@IPROGRAMINCPP:~$
```
Mind you, I've done an update-upgrade cycle with apt so I think it should actually be the latest

**Comment by timschumi** [comment]
Presumably fixed by #14468.
